mlflow/projects/__init__.py

```diff
@@ -169,14 +169,20 @@ def _run(
             build_image=build_image,
             docker_auth=docker_auth,
         )
-        image_digest = kb.push_image_to_registry(image.tags[0])
+        if build_image:
```

Comment:
Maybe it is a good idea to put the `build_docker_image` call that is directly above also in this if-block.
That would make a lot of sense to me.
The `if not build_image:` check can then be removed from `build_docker_image`.
It would extend the scope of the pull request to "Do not push Docker image when --build-image is not specified and also do not build the Docker image"